src/components/navbar/mobile/parts/searchM/SearchRoadmapM.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState } from 'react';
+import { motion, useAnimation } from 'framer-motion';
+import {
+  exploreQueryStore,
+  setExploreQuery,
+} from '@src/components/explore/stores/explore-query-store';
+import { useStore } from '@nanostores/react';
+import { useTriggerRerender } from '@src/hooks/useTriggerRerender';
+import dropclose from '@assets/cross.svg';
+import AnimLoupe from '../animsvg/AnimLoupe';
+
+const SearchRoadmapM = ({
+  handleSearchClick,
+}: {
+  handleSearchClick: () => void;
+}) => {
+  const [inputExpanded, setInputExpanded] = useState(false);
+  const controls = useAnimation();
+  const [isExplorePage, setIsExplorePage] = useState(false);
+  const { params } = useStore(exploreQueryStore);
+  const { query } = params;
+
+  const handleLoupeClick = (e) => {
+    handleSearchClick();
+    e.stopPropagation();
+    setInputExpanded((prev) => !prev);
+    if (inputExpanded) {
+      controls.start({
+        width: '0px',
+        paddingLeft: '0px',
+        paddingRight: '0px',
+        paddingTop: '0px',
+        paddingBottom: '0px',
+      }); // Collapse the input
+    } else {
+      controls.start({
+        width: '250px',
+        paddingLeft: '8px',
```

Comment:
width should be screen width - the size of the other components see [calc](https://developer.mozilla.org/en-US/docs/Web/CSS/calc) this makes the search small on some devices and on other makes it not fit the screen